src/plugin/withIosSourcemaps/withIosSourcemaps.ts

```diff
@@ -7,46 +7,57 @@
 import type { ConfigPlugin } from "@expo/config-plugins";
 import { withXcodeProject } from "@expo/config-plugins";
 
+import { SourceMapUploadOptions } from "../getErrorTrackingPluginsFromOptions";
+
 const SOURCEMAP_FILE_COMMAND =
   "export SOURCEMAP_FILE=$DERIVED_FILE_DIR/main.jsbundle.map";
-const DATADOG_XCODE_COMMAND = `../node_modules/.bin/datadog-ci react-native xcode \`\\\"$NODE_BINARY\\\" --print \\\"require('path').dirname(require.resolve('react-native/package.json')) + '/scripts/react-native-xcode.sh'\\\"\``;
-
-const withIosSourcemaps: ConfigPlugin<void> = (config) => {
-  return withXcodeProject(config, async (config) => {
-    const xcodeProject = config.modResults;
-    const bundlePhase = xcodeProject.pbxItemByComment(
-      "Bundle React Native code and images",
-      "PBXShellScriptBuildPhase"
-    );
-    if (bundlePhase.shellScript.match("datadog-ci react-native xcode")) {
-      return config;
-    }
-    if (bundlePhase.shellScript.match("sentry")) {
-      /**
-       * In the case where Sentry is already added, we add a new line to
-       * bundle the javascript code twice and this time send the sourcemap
-       * to Datadog.
-       * That's the safest way to make sure we don't break the script.
-       *
-       * We also need to remove the final " and add it back for the script to
-       * be well-formatted.
-       */
-      bundlePhase.shellScript = `${bundlePhase.shellScript.replace(
-        /.$/,
-        ""
-      )}\\n ${DATADOG_XCODE_COMMAND}"`;
+const getDatadogXCodeCommand = (
+  serviceName: SourceMapUploadOptions["serviceName"]
+) =>
+  `../node_modules/.bin/datadog-ci react-native xcode \`\\\"$NODE_BINARY\\\"${
+    serviceName ? ` --service ${serviceName}` : ""
+  } --print \\\"require('path').dirname(require.resolve('react-native/package.json')) + '/scripts/react-native-xcode.sh'\\\"\``;
```

Comment:
Here you're applying the `--service` argument to the node command that's resolving the path to react-native, so it won't be applied to your sourcemaps upload.
To pass it to the `datadog-ci` command you need to apply it at the very end of the command.

```suggestion
  `../node_modules/.bin/datadog-ci react-native xcode \`\\\"$NODE_BINARY\\\" --print \\\"require('path').dirname(require.resolve('react-native/package.json')) + '/scripts/react-native-xcode.sh'\\\"\` ${
    serviceName ? ` --service ${serviceName}` : ""
  } `;
```

Reply:
Updated! Thanks for calling this out. At the moment I don't have a datadog project to be able to test this with. I was utilizing the tests to make sure the command was structured correctly.

Would you happen to have a test project that we could run this with to verify its configured properly?